### DISCUSSION TITLE
ra-tls-secret-prov demo： error: AESM service returned error 1; this may indicate that infrastructure for the DCAP attestation requested by Gramine is missing on this machine

### DISCUSSION THREAD

**Discussion by RicLee0124** [gramineproject/gramine#971] [Q&A]
i am testing tihis demo https://github.com/gramineproject/gramine/tree/v1.3/CI-Examples/ra-tls-secret-prov, I test as the following code:
make app dcap RA_TYPE=dcap
cd secret_prov_pf
RA_TLS_ALLOW_DEBUG_ENCLAVE_INSECURE=1 \
RA_TLS_ALLOW_OUTDATED_TCB_INSECURE=1 \
./server_dcap wrap_key &

gramine-sgx ./client

when I run "gramine-sgx ./client", I got an error:
 "error: AESM service returned error 1; this may indicate that infrastructure for the DCAP attestation requested by Gramine is missing on this machine
error: load_enclave() failed with error -1"

I think I have not instlalled the dcap attestation infrastructure, so what shoud i intall on gramine.
additional: I use "sudo apt-get install gramine"  to instlall gramine following thie step https://gramine.readthedocs.io/en/latest/quickstart.html

**Comment by dimakuv** [comment]
> I think I have not instlalled the dcap attestation infrastructure

1. If you're using a Microsoft Azure Confidential Computing VM, then things are simple, you just need to install the `az-dcap-client` package. See https://learn.microsoft.com/en-us/azure/attestation/faq#how-can-a-verifier-obtain-the-collateral-for-sgx-attestation-supported-by-azure-attestation

2. If you're using your own local machine, then you need to install the DCAP attestation infrastructure locally. Be aware: **this is tedious**. There is a document on how to install it: https://download.01.org/intel-sgx/sgx-dcap/1.14/linux/docs/Intel_SGX_SW_Installation_Guide_for_Linux.pdf
